File: Easy/Python/2108. Find First Palindromic String in the Array.py
class Solution(object):
    def firstPalindrome(self, words):
        for s in words:
            left = 0
            right = len(s) - 1
            isPalindrome = True

            # Use 2 pointers to compare each character in each word
            while left <= right:
                if s[left] != s[right]:
                    isPalindrome = False
                left += 1
                right -= 1
            
            # If the word is a palindrome, return it
            if(isPalindrome):
                return s
        
        return ""

File: Easy/Java/2108. Find First Palindromic String in the Array.java
class Solution {
    public String firstPalindrome(String[] words) {
        for(String s : words){
            int left = 0;
            int right = s.length() - 1;
            boolean isPalindrome = true;
            
            // Use 2 pointers to compare each character in each word
            while(left <= right){
                if(s.charAt(left) != s.charAt(right))
                    isPalindrome = false;
                left++;
                right--;
            }

            // If the word is a palindrome, return it
            if(isPalindrome)
                return s;
        }

        return "";
    }
}
